pkg/api/client/client.go

```diff
@@ -0,0 +1,111 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/bmc-toolbox/common"
+	rivets "github.com/metal-toolbox/rivets/types"
+)
+
+const (
+	versionEndpoint            = "/api/version"
+	componentsEndpoint         = "/components"
+	inbandInventoryEndpoint    = "/inventory/in-band"
+	outofbandInventoryEndpoint = "/inventory/out-of-band"
+)
+
+type ServerComponents map[string][]*rivets.Component
+
+// Client can perform queries against the Component Inventory Service.
+type Client interface {
+	Version(context.Context) (string, error)
+	GetServerComponents(context.Context, string) (ServerComponents, error)
+	UpdateInbandInventory(context.Context, string, *common.Device) (string, error)
+	UpdateOutOfbandInventory(context.Context, string, *common.Device) (string, error)
+}
+
+type componentInventoryClient struct {
+	// The server address with the schema
+	serverAddress string
+	// Authentication token
+	authToken string
+	// Doer for performing requests, typically a *http.Client with any
+	// customized settings, such as certificate chains.
+	client httpRequestDoer
+}
+
+// Creates a new Client, with reasonable defaults
+func NewClient(serverAddress string, opts ...Option) (Client, error) {
+	// create a client with sane default values
+	client := componentInventoryClient{serverAddress: serverAddress}
+	// mutate client and add all optional params
+	for _, o := range opts {
+		if err := o(&client); err != nil {
+			return nil, err
+		}
+	}
+
+	// create httpClient, if not already present
+	if client.client == nil {
+		client.client = &http.Client{}
+	}
+
+	return client, nil
+}
+
+func (c componentInventoryClient) GetServerComponents(ctx context.Context, serverID string) (ServerComponents, error) {
```

Comment:
This is fine for the current iteration. I wonder if there is something in `bmc-toolbox/common` that would be better than `rivets`. I'll look into that today.

Reply:
Thanks!
It looks `common.Device` can also fit here. Do we want to discard `rivets` in favor of `bmc-toolbox/common`?